src/Harmony-One/Signer.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Signer.h"
+#include "../Ethereum/RLP.h"
+#include "../HexCoding.h"
+
+using namespace TW;
+using namespace TW::Harmony;
+
+std::tuple<uint256_t, uint256_t, uint256_t> Signer::values(const uint256_t &chainID,
+                                                           const Data &signature) noexcept {
+    boost::multiprecision::uint256_t r, s, v;
+    import_bits(r, signature.begin(), signature.begin() + 32);
+    import_bits(s, signature.begin() + 32, signature.begin() + 64);
+    import_bits(v, signature.begin() + 64, signature.begin() + 65);
+    v += 27;
+
+    boost::multiprecision::uint256_t newV;
+    import_bits(newV, signature.begin() + 64, signature.begin() + 65);
+    newV += 35 + chainID + chainID;
+    return std::make_tuple(r, s, newV);
+}
```

Comment:
did you mix two implementation into one? the previous r,s,v is not used

Reply:
No, since we are only going with EIP155 based logic for the future, hence no need to do the branching logic that you might be referring to.